cruise_company_spring/src/main/java/com/example/cruise_company/service/impl/RouteServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package com.example.cruise_company.service.impl;
+
+import com.example.cruise_company.controller.dto.RouteDto;
+import com.example.cruise_company.service.RouteService;
+import com.example.cruise_company.service.model.Route;
+import com.example.cruise_company.service.repository.PortRepository;
+import com.example.cruise_company.service.repository.RouteRepository;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
```

Comment:
Somewhere you are using @Component, somewhere @Service for service classes, please use @Service annotation in each service implementation class